fontc/src/main.rs

```diff
@@ -203,52 +225,66 @@ fn do_work(work: Vec<Box<dyn Work>>) -> Result<(), Error> {
     Ok(())
 }
 
-struct CompileContext {
+struct ChangeDetector {
     paths: Paths,
     ir_source: Box<dyn Source>,
     prev_inputs: Input,
     current_inputs: Input,
 }
 
-impl CompileContext {
-    fn new(paths: Paths, args: Args) -> Result<CompileContext, Error> {
+impl ChangeDetector {
+    fn new(paths: Paths, args: Args) -> Result<(Config, ChangeDetector), Error> {
```

Comment:
i find it odd that new (supposedly the default constructor of an object?) returns a tuple instead of just the newly initialized object...

Reply:
i suggest you initialise the Config beforehand from args, and pass that to ChangeDetector::new

Reply:
+1, fixing

Reply:
Fixed